src/components/AssetManagementPanel/AssetManagementForm/FormVariants/EndorseBeneficiary/EndorseBeneficiary.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useContext, useEffect, useState } from "react";
+import { FormState } from "../../../../../constants/FormState";
+import { ButtonSolidOrangeWhite, ButtonSolidWhiteGrey } from "../../../../UI/Button";
+import { LoaderSpinner } from "../../../../UI/Loader";
+import { AssetInformationPanel } from "../../../AssetInformationPanel";
+import { AssetManagementActions } from "../../../AssetManagementActions";
+import { AssetManagementTitle } from "../../AssetManagementTitle";
+import { OverlayContext } from "./../../../../../common/contexts/OverlayContext";
+import {
+  MessageTitle,
+  showDocumentTransferMessage,
+} from "./../../../../../components/UI/Overlay/OverlayContent/DocumentTransferMessage";
+import { EditableAssetTitle } from "./../EditableAssetTitle";
+
+interface EndorseBeneficiaryProps {
+  formAction: AssetManagementActions;
+  tokenId: string;
+  tokenRegistryAddress: string;
+  beneficiary?: string;
+  holder?: string;
+  handleTransfer: (newBeneficiary: string, newHolder: string) => void;
+  beneficiaryEndorseState: string;
+  onBack: () => void;
+}
+
+export const EndorseBeneficiaryForm = ({
+  formAction,
+  tokenId,
+  tokenRegistryAddress,
+  beneficiary,
+  holder,
+  handleTransfer,
+  beneficiaryEndorseState,
+  onBack,
+}: EndorseBeneficiaryProps) => {
+  const [newBeneficiary, setNewBeneficiary] = useState("");
+  const [newHolder, setNewHolder] = useState("");
+  const isPendingConfirmation = beneficiaryEndorseState === FormState.PENDING_CONFIRMATION;
+  const isConfirmed = beneficiaryEndorseState === FormState.CONFIRMED;
+  const isEditable =
+    beneficiaryEndorseState !== FormState.PENDING_CONFIRMATION && beneficiaryEndorseState !== FormState.CONFIRMED;
+  const { showOverlay } = useContext(OverlayContext);
+
+  useEffect(() => {
+    if (isConfirmed) {
+      showOverlay(
+        showDocumentTransferMessage(MessageTitle.CHANGE_BENEFICIARY_SUCCESS, {
+          isSuccess: true,
+          beneficiaryAddress: newBeneficiary,
+        })
+      );
+    }
+  }, [isConfirmed, newBeneficiary, showOverlay]);
+
+  const isValidEndorse = () => {
+    if (!newBeneficiary || !newHolder) return false;
+    if (newBeneficiary === beneficiary && newHolder === holder) return false;
```

Comment:
might be weird if you fail silently here

Reply:
user who tried it wont know this condition

Reply:
ok noted